### PR TITLE
YLP-1892 added integration test for empty field

### PR DESCRIPTION
**Pull request by Xwdrblue**

### CODE REVIEW

File: packages/yourloops/test/integration/pages/home-page/hcp-home-page.spec.tsx [hunk 1]
```diff
@@ -216,4 +217,58 @@ describe('HCP home page', () => {
 
     expect(invitePatientButton).toBeEnabled()
   })
+
+  it('should check that the team creation button is valid if all fields are complete', async () => {
+    await act(async () => {
+      render(getHomePage())
```
Comment (TimCanu) [inline packages/yourloops/test/integration/pages/home-page/hcp-home-page.spec.tsx:223]:
```suggestion
      renderPage('/')
```

renderPage import: `import { renderPage } from '../../utils/render'`

File: packages/yourloops/test/integration/pages/home-page/hcp-home-page.spec.tsx [hunk 2]
```diff
@@ -216,4 +217,58 @@ describe('HCP home page', () => {
...
+    })
+    const teamMenu = screen.getByLabelText('Open team menu')
+    userEvent.click(teamMenu)
+    userEvent.click(screen.getByText('New care team'))
+    const dialogTeam = screen.getByRole('dialog')
+    const createTeamButton = within(dialogTeam).getByRole('button', { name: 'Create team' })
+    const nameInput = within(dialogTeam).getByRole('textbox', { name: 'Name' })
+    const adress1Input = within(dialogTeam).getByRole('textbox', { name: 'Address 1' })
```
Comment (TimCanu) [inline packages/yourloops/test/integration/pages/home-page/hcp-home-page.spec.tsx:231]:
```suggestion
    const address1Input = within(dialogTeam).getByRole('textbox', { name: 'Address 1' })
```

File: packages/yourloops/test/integration/pages/home-page/hcp-home-page.spec.tsx [hunk 3]
```diff
@@ -216,4 +217,58 @@ describe('HCP home page', () => {
...
+    const adress2Input = within(dialogTeam).getByRole('textbox', { name: 'Address 2' })
```
Comment (TimCanu) [inline packages/yourloops/test/integration/pages/home-page/hcp-home-page.spec.tsx:232]:
```suggestion
    const address2Input = within(dialogTeam).getByRole('textbox', { name: 'Address 2' })
```

File: packages/yourloops/test/integration/pages/home-page/hcp-home-page.spec.tsx [hunk 4]
```diff
@@ -216,4 +217,58 @@ describe('HCP home page', () => {
...
+    await userEvent.type(adress1Input, '5 rue')
+    await userEvent.type(zipcodeInput, '38000')
+    await userEvent.type(cityInput, 'Grenoble')
+    await userEvent.type(phoneNumberInput, '0600000000')
+    await userEvent.type(emailInput, 'toto@titi.com')
+    expect(createTeamButton).not.toBeDisabled()
+
+    // Team creation button disabled and a field in error with the error message
+    userEvent.clear(phoneNumberInput)
+    userEvent.clear(emailInput)
```
Comment (TimCanu) [inline packages/yourloops/test/integration/pages/home-page/hcp-home-page.spec.tsx:267]: You can also clear the zip code and test the validation on it?

File: packages/yourloops/test/integration/pages/home-page/hcp-home-page.spec.tsx [hunk 5]
```diff
@@ -216,4 +217,58 @@ describe('HCP home page', () => {
...
+    await userEvent.type(phoneNumberInput, '060')
+    expect(within(dialogTeam).getByText('Please enter a valid phone number')).toBeVisible()
+    await userEvent.type(emailInput, 'tototiti.com')
+    expect(within(dialogTeam).getByText('Invalid email address (special characters are not allowed).')).toBeVisible()
+    expect(createTeamButton).toBeDisabled()
```
Comment (TimCanu) [inline packages/yourloops/test/integration/pages/home-page/hcp-home-page.spec.tsx:272]: I would then fill the phone number, the email and the zip code properly, check that the button is enabled, click on the create button and check that the API call has been made :slightly_smiling_face:

Comment (Xwdrblue) [reply packages/yourloops/test/integration/pages/home-page/hcp-home-page.spec.tsx:272]: Okay but this part is made to test the cases where the inputs of the form are incorrect and with the disable button

Comment (Xwdrblue) [reply packages/yourloops/test/integration/pages/home-page/hcp-home-page.spec.tsx:272]: I then add above the test with the API call

Comment (TimCanu) [reply packages/yourloops/test/integration/pages/home-page/hcp-home-page.spec.tsx:272]: If it is easily possible, it is better to do everything in one test. The more tests we have, the longer it takes (as you have to initialize the DOM in each test)

File: packages/yourloops/test/integration/pages/home-page/hcp-home-page.spec.tsx [hunk 6]
```diff
@@ -216,4 +219,80 @@ describe('HCP home page', () => {
...
+
+    userEvent.clear(emailInput)
+    await userEvent.type(emailInput, 'tototiti.com')
+    expect(within(dialogTeam).getByText('Invalid email address (special characters are not allowed).')).toBeVisible()
+    expect(createTeamButton).toBeDisabled()
+    userEvent.clear(emailInput)
+    await userEvent.type(emailInput, 'toto@titi.com')
+    expect(createTeamButton).not.toBeDisabled()
+
+    userEvent.click(createTeamButton)
```
Comment (TimCanu) [inline packages/yourloops/test/integration/pages/home-page/hcp-home-page.spec.tsx:295]:
```suggestion
    await act(async () => {
      userEvent.click(createTeamButton)
    })
```
Without this. You have error in the console. I suggest you try it locally to see (with and without). If you don't understand this fix I can explain it to you

Comment (Xwdrblue) [reply packages/yourloops/test/integration/pages/home-page/hcp-home-page.spec.tsx:295]: Without your correction I have the tests on the glass and when I add 
await act(async () => {
      userEvent.click(createTeamButton)
    }) the tests are red

Comment (TimCanu) [reply packages/yourloops/test/integration/pages/home-page/hcp-home-page.spec.tsx:295]: Are you sure? It is working on my side

**Review by TimCanu — COMMENTED** [review]
Add `YLP-1892 added integration test for empty field` in the engineering part of the changelog and we are good :slightly_smiling_face:
